src/stepfunctions/steps/utils.py

```diff
@@ -45,3 +46,24 @@ def get_aws_partition():
             return cur_partition
 
     return cur_partition
+
+
+def merge_dicts(first, second, first_name, second_name):
```

Comment:
This could also be used to merge the [hyperparameters](https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/src/stepfunctions/steps/sagemaker.py#L175-#L180) in TrainingStep - will make the changes in another PR

Reply:
suggestion: First and second don't describe the side effects and which dict gets merged into what. Borrowing from [JavaScript's `Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#syntax):

```suggestion
def merge_dicts(target, source):
```